File: Ruby/008_string_char_occurrence.rb
######################################################################
# this file contains different implementations of counting how many
# times character appears in string
######################################################################

# this method uses built-in function 'scan'
def char_counter_1(string, char)
  string.scan(/#{char}/).length
end

# this method uses built-in function 'count'
def char_counter_2(string, char)
  string.count(char)
end

# this method uses built-in function 'split'
def char_counter_3(string, char)
  counter = 0
  string.split('').each do |key|
    if key == char
        counter += 1
    end
  end
  counter
end

string = 'I have a string with multiple characters'
char = 'a'
puts char_counter_1(string, char)
puts char_counter_2(string, char)
puts char_counter_3(string, char)

File: Python/008_string_char_occurrence.py
######################################################################
# this file contains different implementations of counting how many
# times character appears in string
######################################################################


# this method uses built-in 'count' function
def char_counter_1(string, char):
    return string.count(char)


def char_counter_2(string, char):
    counter = 0
    for key in string:
        if key == char:
            counter += 1
    return counter


def main():
    string = 'I have a string with multiple characters'
    char = 'a'
    print(char_counter_1(string, char))
    print(char_counter_2(string, char))

if __name__ == '__main__':
    main()